client/engine/engine.go

```diff
@@ -86,25 +86,34 @@ func (e *Engine) ToApi() <-chan ObjectiveChangeEvent {
 func (e *Engine) Run() {
 	for {
 		var res ObjectiveChangeEvent
+		var err error
 		select {
 		case apiEvent := <-e.FromAPI:
-			res = e.handleAPIEvent(apiEvent)
+			res, err = e.handleAPIEvent(apiEvent)
 
 		case chainEvent := <-e.fromChain:
 
-			res = e.handleChainEvent(chainEvent)
+			res, err = e.handleChainEvent(chainEvent)
 
 		case message := <-e.fromMsg:
-			res = e.handleMessage(message)
+			res, err = e.handleMessage(message)
+		}
 
+		// Handle errors
+		if err != nil {
+			panic(err)
```

Comment:
We could use `e.logger.Panic()` here so the error also get's written out to the log.

Reply:
[d72d90c](https://github.com/statechannels/go-nitro/pull/402/commits/d72d90cf0617e3f06b6c9801013ab08f997736c1)